### PR TITLE
Comments and launchdarkly

### PR DESCRIPTION
**Pull request by skow0020**
This pull request introduces entry comments as a new feature, protected by a LaunchDarkly feature flag for controlled rollout. It adds the necessary database schema, backend logic, frontend UI, and comprehensive tests to support commenting on journal entries. The feature is toggleable in non-production environments for testing purposes, and includes documentation updates.

**Entry Comments Feature Implementation**

* Added a new `entry_comments` table with foreign key constraints and indexes to the database schema to store comments on journal entries. [[1]](diffhunk://#diff-bff5e43549cdacd09b259c00ef467e60a37553a4e4e656f393f16a56477c6566R1-R12) [[2]](diffhunk://#diff-349e2e5efe6c282645244197ec3a36bee4f6fbc9988ee5acc46a05603f6829cdR18-R24)
* Implemented the `addCommentAction` server action, including input validation, permission checks, LaunchDarkly feature flag evaluation, and integration with the new data helper for comment creation. ([src/app/dashboard/journals/[journalId]/actions.tsR73-R83](diffhunk://#diff-bb49da40a2e37f5ac92b3365f32c55cbc17b43af0e4d414188db6ce20d0490faR73-R83), [src/app/dashboard/journals/[journalId]/actions.tsR155-R160](diffhunk://#diff-bb49da40a2e37f5ac92b3365f32c55cbc17b43af0e4d414188db6ce20d0490faR155-R160), [src/app/dashboard/journals/[journalId]/actions.tsR444-R502](diffhunk://#diff-bb49da40a2e37f5ac92b3365f32c55cbc17b43af0e4d414188db6ce20d0490faR444-R502))

**Feature Flag Integration and Testing**

* Integrated LaunchDarkly for feature flagging the entry comments functionality, including environment variable setup, a server-side helper, and new API endpoints for toggling the flag in test environments. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R24) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R30-R35) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R127-R155) [[4]](diffhunk://#diff-78741b88ba788cdf515649a51b4927fe4dd1322def54bfb27d062c8631ddc1cbR1-R27)
* Updated end-to-end and unit tests to cover the new commenting workflow, feature flag toggling, and all relevant edge cases. [[1]](diffhunk://#diff-483cb85dc45d0c52f56a526f077fbbbb5318ff0dafc61b2e4a3d3dedd1799d8aL5-R27) [[2]](diffhunk://#diff-483cb85dc45d0c52f56a526f077fbbbb5318ff0dafc61b2e4a3d3dedd1799d8aR56-R59) [[3]](diffhunk://#diff-7d5de5f497ee57f163d0f307ce4b752d624744069d286290d22ed9acb7e4d0e2R95-R107) [src/app/dashboard/journals/[journalId]/actions.test.tsR6](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R6), [src/app/dashboard/journals/[journalId]/actions.test.tsR17-R21](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R17-R21), [src/app/dashboard/journals/[journalId]/actions.test.tsR32](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R32), [src/app/dashboard/journals/[journalId]/actions.test.tsR48-R51](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R48-R51), [src/app/dashboard/journals/[journalId]/actions.test.tsR78-R84](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R78-R84), [src/app/dashboard/journals/[journalId]/actions.test.tsR96-R97](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R96-R97), [src/app/dashboard/journals/[journalId]/actions.test.tsR640-R739](diffhunk://#diff-58a963f7de38b53f9f847c6b192667bd1ed9057605936ac1fde41be206945cb2R640-R739), [src/app/dashboard/journals/[journalId]/entry-comments.test.tsxR1-R118](diffhunk://#diff-82dbceea0e612104eb7ec68e52ae15cc2c4ae8053056b8ca9602b4cfa8b4a1d1R1-R118))

**Documentation**

* Updated the `README.md` with instructions for setting up LaunchDarkly, including environment variables and usage examples. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R30-R35) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R127-R155)

Let me know if you have questions about the feature flag integration or the new comment-related logic!

### CODE REVIEW

File: src/data/comments.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }).returning()
+
+  const decryptedContent = decryptEntryContent(comment.content)
+
+  return comment
+    ? {
+        ...comment,
+        content: decryptedContent,
+      }
+    : null
```
[the remark this answers is not in the document]
Comment (skow0020) [reply src/data/comments.ts]: Done

File: src/data/comments.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+// --- Types ---
+export const CreateEntryCommentSchema = z.object({
+  entryId: z.string().uuid(),
+  authorUserId: z.string().uuid(),
+  content: z.string().trim().min(1).max(2000),
+})
+export type CreateEntryCommentInput = z.infer<typeof CreateEntryCommentSchema>
+
+export async function createEntryComment(input: CreateEntryCommentInput) {
```
[the remark this answers is not in the document]
Comment (skow0020) [reply src/data/comments.ts]: Done
